lib/scripting/post-process/expr.ts

```diff
@@ -17,138 +17,288 @@
+function ppLogicalExpression(node: ESIToken): any {
+	let estree = node.children[0].estree;
+	let index = node.children[0].text.length;
+	for (let loop = 1; loop < node.children.length; loop++) {
+		if (node.text.substr(index).startsWith(' And ')) {
+			estree = logicalExpression('&&', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 5;
+		} else if (node.text.substr(index).startsWith(' Or ')) {
+			estree = logicalExpression('||', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 4;
+		} else if (node.text.substr(index).startsWith(' Xor ')) {
+			estree = logicalExpression(
+				'||',
+				logicalExpression('&&', estree, unaryExpression('!', node.children[loop].estree)),
+				logicalExpression('&&', unaryExpression('!', estree), node.children[loop].estree),
+			);
+			index += node.children[loop].text.length + 5;
+		} else if (node.text.substr(index).startsWith(' Eqv ')) {
+			estree = unaryExpression('~', binaryExpression('^', estree, node.children[loop].estree));
+			index += node.children[loop].text.length + 5;
+		}
+	}
+	return estree;
 }
 
-export function add(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression(operator, leftExpr, rightExpr);
+function ppRelationalExpression(node: ESIToken): any {
+	let estree = node.children[0].estree;
+	let index = node.children[0].text.length;
+	for (let loop = 1; loop < node.children.length; loop++) {
+		if (node.text.substr(index).startsWith('<>')) {
+			estree = binaryExpression('!=', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 2;
+		} else if (node.text.substr(index).startsWith('<=')) {
+			estree = binaryExpression('<=', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 2;
+		} else if (node.text.substr(index).startsWith('>=')) {
+			estree = binaryExpression('>=', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 2;
+		} else if (node.text.substr(index).startsWith('=')) {
+			estree = binaryExpression('==', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 1;
+		} else {
+			estree = binaryExpression(node.text.charAt(index) as BinaryOperator, estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 1;
+		}
+	}
+	return estree;
 }
 
-export function mod(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('%', leftExpr, rightExpr);
+function ppTypeExpression(node: ESIToken): any {
+	let estree = node.children[0].estree;
+	let index = node.children[0].text.length;
+	for (let loop = 1; loop < node.children.length; loop++) {
+		if (node.text.substr(index).startsWith(' Is ')) {
```

Comment:
should this be case insensitive?

Reply:
i think you could also build a look up table, then you could rewrite ppRelationalExpression much simpler!

Reply:
No. When we reformat the script, we normalize all keywords.